src/sagas.js

```diff
@@ -295,19 +402,106 @@ function* schedulerPerMinute (action) {
   }
 }
 
+function* createNewVm (action) {
```

Comment:
again, I think it's much better to wrap this logic into a React component and dispatch a "CreateVM" action once the "Submit" button is clicked.